# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatform.cpp
# Commit: 887e5358bad0
# Full Hash: 887e5358bad02aa565b0a491ac97644f6bf32f53
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-19 21:34:15
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Don't force sync for async-present if GL not thread-safe. r=aosmond a=graft
#   
#   If GL is not thread-safe, then the check for whether we use the canvas render thread
#   is not accurate. The thread-safe check takes precedence, and if we are not thread-safe,
#   we are actually on the render thread. It should be safe to not force sync here since
# ==============================================================================

diff -r 3fd71c45d9fc -r 887e5358bad0 gfx/thebes/gfxPlatform.cpp
--- a/gfx/thebes/gfxPlatform.cpp	Tue Dec 19 17:25:55 2023 +0200
+++ b/gfx/thebes/gfxPlatform.cpp	Tue Dec 19 20:44:27 2023 +0000
@@ -3046,7 +3046,7 @@
   gfxVars::SetUseCanvasRenderThread(feature.IsEnabled());
 
   bool webglOopAsyncPresentForceSync =
-      !gfxVars::UseCanvasRenderThread() ||
+      (threadsafeGL && !gfxVars::UseCanvasRenderThread()) ||
       StaticPrefs::webgl_out_of_process_async_present_force_sync();
   gfxVars::SetWebglOopAsyncPresentForceSync(webglOopAsyncPresentForceSync);
 